redash/models/__init__.py

```diff
@@ -630,34 +638,35 @@ def outdated_queries(cls):
         scheduled_queries_executions.refresh()
 
         for query in queries:
-            if query.schedule["interval"] is None:
-                continue
+            try:
+                if query.schedule["until"]:
+                    schedule_until = pytz.utc.localize(
+                        datetime.datetime.strptime(query.schedule["until"], "%Y-%m-%d")
+                    )
 
-            if query.schedule["until"] is not None:
-                schedule_until = pytz.utc.localize(
-                    datetime.datetime.strptime(query.schedule["until"], "%Y-%m-%d")
+                    if schedule_until <= now:
+                        continue
+
+                retrieved_at = scheduled_queries_executions.get(query.id) or (
+                    query.latest_query_data and query.latest_query_data.retrieved_at
                 )
 
-                if schedule_until <= now:
-                    continue
-
-            if query.latest_query_data:
-                retrieved_at = query.latest_query_data.retrieved_at
-            else:
-                retrieved_at = now
-
-            retrieved_at = scheduled_queries_executions.get(query.id) or retrieved_at
-
-            if should_schedule_next(
-                retrieved_at,
-                now,
-                query.schedule["interval"],
-                query.schedule["time"],
-                query.schedule["day_of_week"],
-                query.schedule_failures,
-            ):
-                key = "{}:{}".format(query.query_hash, query.data_source_id)
-                outdated_queries[key] = query
+                if should_schedule_next(
+                    retrieved_at or now,
+                    now,
+                    query.schedule["interval"],
+                    query.schedule["time"],
+                    query.schedule["day_of_week"],
+                    query.schedule_failures,
+                ):
+                    key = "{}:{}".format(query.query_hash, query.data_source_id)
+                    outdated_queries[key] = query
+            except Exception as e:
+                logging.info(
+                    "Could not determine if query %d is outdated due to %s",
+                    query.id,
+                    repr(e),
+                )
```

Comment:
1. This should go into Sentry.
2. Considering the way our scheduler works this is going to be spammy.  Maybe we should report this and then unschedule?

Reply:
Regarding #2 - this function should be idempotent. It's kinda strange to call `outdated_queries` and have your schedule change as a result (perhaps it was a volatile error?).

I feel more comfortable with calling `track_failure` in these cases - this way users could find out that there was an error with running the query, and it would increase the `schedule_failures` counter, which would push the next reschedule down the exponential backoff track.

Reply:
Actually it would make more sense to use `track_failure` in `_enqueue_queries` instead.

Reply:
I don't think `schedule_failures` will work here, because it won't reach the part of the code that calculates the next iteration.

While usually you're right in the approach, we're talking about queries with malformed schedule object. Also, sending a user a notification about it doesn't feel useful as in most cases they won't know what to do about it.

Another approach we can take is to introduce a `disabled` boolean to the `schedule` object which we will trigger in such cases (and later can review these). The downside here is that it won't work in case it's completely malformed (not even a valid JSON).

Reply:
well, `disabled` could be set to `True` if `schedule` is broken or `schedule['disabled']` is `True`.

Reply:
👍

Reply:
Reporting to Sentry is done in 57b1d4f.